keps/1714-fair-sharing/README.md

```diff
@@ -0,0 +1,366 @@
+resources (having workloads needing more than nominal quota), should 
+be given a similar amount of resources. And any pre existing imbalances should
+be timely resolved by preemption.
+
+* Allow system admin to indicate that a sub-organization is more important than
+their siblings, and should get proportionally more resources than others located
+under the same parent. 
+
+* When distributing resources, prioritize needs of CQ within the same
+sub-organization and only after they have been fulfilled, proceed with distribution
+outside of the suborganization.
+
+* When enforcing fair sharing, ignore workload priorities unless:
+
+** The workload's priority is above admin-defined high priority. Super high
+priority workloads overrule fair sharing and are treated according to KEP #1337.
+
+** There is a need to preempt some non top priority workload from a ClusterQueue.
+Then the lowest priority workloads from a CQ that is over its fair share should
+be selected for preemption.
+
+* The mechanism should be compatible with all existing or proposed Kueue features,
+in particular:
+
+** Borrowing limits 
+** Lending limits
+** Guaranteed/nominal quota
+** Hierarchical cohorts
+
+* Fair sharing should not limit Kueue scalability. Kueue, with fair sharing enabled,
+should be able to handle >1k ClusterQueues, >100 Cohorts and >10k workloads (that
+are either running or queued) within a single hierarchical organization.
+
+* The proposed system should be hard to game, for example by creating big workloads
+that consume all capacity.
+
+* Fair sharing enforcement should not significantly decrease overall utilization,
+however, pathological situations (like a single workload consuming all otherwise
+unused capacity) should be resolved in favor of fair sharing than maximizing the
+utilization (the big greedy workload should be preempted to admit smaller
+workloads from other CQ that consume only their fair share).
+
+### Non-Goals
+* Use historical data (for example CQ A used a lot of shared capacity for the last
+week, so now it should get less because others, who didn't need anything then, have
+pending workloads). Fair sharing should be based on point-in-time situation, although,
+ideally it should be expandable to support history-based fair sharing without major redesign.
+
+* Enable fair sharing only for some part of the resources or Cohort hierarchy.
+Fair sharing will be a global switch (at least initially).
+
+* Maximize utilization at the cost of fair sharing.
+
+## Proposal
+Introduce a global fair sharing mechanism that is based on preemptions. As long
+as there are some free and accessible resources in the cohort hierarchy, Kueue
+will admit workloads without any limits. However, once the capacity is gone,
+new workloads from Cohorts/CQ that have not received their fair share yet will
+be permitted to preempt workloads from Chorots/CQ that received more than their
+fair share would otherwise be (or more than the Cohort/CQ that is in need).
```

Comment:
Can we define at this point what "fair share" is? In particular, is it proportional to "nominalQuota", what is the unit of the values?

I assume it is defined later, but it seems reasonable to define it early, not necessarily fully formally, so that the reader has some intuition what this is already. If not feasible to define here maybe at least reference with the line to the place where the concept is defined later.

Reply:
Added.